src/pages/ProductListing/ProductListing.jsx

```diff
@@ -1,10 +1,131 @@
-import {Route,Link,Routes} from "react-router-dom"
+import * as All from "../../assets/icons/icons.jsx";
+import { useEffect, useReducer, useState } from "react";
+import axios from "axios";
+import "./productListing.css"
+import { useCategory } from "../../context/ProductContext";
+// import { useFilter } from "../../context/FilterContext";
+import { filterProducts } from "../../context/Functions/filterProducts.jsx";
+
 export function ProductListing(){
+    const {category}=useCategory();
+    const [state,dispatch]=useReducer(filterProducts,{items:[],low_to_High:false,High_to_low:false,tshirt:false,bobble:false,badges:false,sticker:false,rating_number:"1"});
```

Comment:
you are doing is great but I have a suggestion for you, instead of writing initial values in the reducer hook, you can just extract that details in a variable and pass that variable to the reducer hook. It helps your code be clean and readable.

Reply:
Thanks for the suggestion😊